master/internal/api_model.go

```diff
@@ -22,19 +22,50 @@ import (
 	structpb "github.com/golang/protobuf/ptypes/struct"
 )
 
-func (a *apiServer) GetModel(
-	_ context.Context, req *apiv1.GetModelRequest) (*apiv1.GetModelResponse, error) {
+func (a *apiServer) ModelFromIdentifier(identifier string) (*modelv1.Model, error) {
+	var err error
 	m := &modelv1.Model{}
-	switch err := a.m.db.QueryProto("get_model", m, req.ModelName); err {
+
+	allNumbers, _ := regexp.MatchString("^\\d+$", identifier)
+	if allNumbers {
+		err = a.m.db.QueryProto("get_model_by_id", m, identifier)
+	} else {
+		err = a.m.db.QueryProto("get_model", m, identifier)
+	}
+	switch err {
 	case db.ErrNotFound:
 		return nil, status.Errorf(
-			codes.NotFound, "model \"%s\" not found", req.ModelName)
+			codes.NotFound, "model \"%s\" not found", identifier)
+	default:
+		return m, errors.Wrapf(err,
+			"error fetching model \"%s\" from database", identifier)
```

Comment:
what is the reason that you have different error handling in this final default case between ModelFromIdentifier ModelVersionFromID?